modules/api/src/main/java/org/apache/ignite/network/ClusterNode.java

```diff
@@ -32,11 +32,10 @@ public interface ClusterNode extends Serializable {
     String id();
 
     /**
-     * Returns the unique name (consistent ID) of the node in the cluster. Does not change between restarts.
-     *
-     * @return Unique name of a cluster member.
+     * Returns the unique name (consistent ID) of the node in the cluster, {@code null} if the node has not been added to the topology.
+     * Does not change between restarts.
      */
-    String name();
+    @Nullable String name();
```

Comment:
Since this class is in `api` module, and user can only get an instance of a node _with_ name, I think we should remove the annotation from this particular place and revert the javadoc change. Same should apply to the subclasses that cannot in fact have null as a node name, which should be most of subclasses.

Maybe I'm misunderstanding something, please comment in response

Reply:
Thank you for pointing this out, I didn’t pay attention, I’ll return the changes related to this.